### PR TITLE
feat(datepicker): support HTML5 input types

### LINKED ISSUE #3312: Selected date is undefined with HTML5 datepicker and Angular 1.3.x
Using Boostrap UI together with any Angular 1.3.x will break compatibility with HTML5 date pickers. Once the date is selected, it will will set the value in the ng-model for the input field to undefined. This works perfectly in Angular 1.2.x.
[Working sample with Angular 1.2.28](http://plnkr.co/edit/b8BvCLDCPhjQkRlV0x85?p=preview)
[Broken sample with Angular 1.3.0](http://plnkr.co/edit/KFgfhX507jnVeiObnyE3?p=preview)

### PR DESCRIPTION
**Pull request by chrisirhc**
- [x] Add support for other HTML5 input types such as month and datetime-local

Fixes #3312

Demo of this:
http://plnkr.co/edit/eBtCPhNa7S1MnPbUY70a?p=preview

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -580,24 +595,25 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
...
+           return ngModel.$isEmpty(value) ? value : dateFilter(value, dateFormat);
+        });
+      }
 
       // Inner change
       scope.dateSelection = function(dt) {
         if (angular.isDefined(dt)) {
           scope.date = dt;
         }
-        if (dateFormat) {
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Does removing this not break anything?

Comment (chrisirhc) [inline src/datepicker/datepicker.js]: Hm.. Good point, the directive should throw immediately if dateFormat is falsey. It's undefined behavior and definitely an error in the use of this directive.

**Comment by wesleycho** [comment]
Closed by accident, reverted my commit.

**Comment by wesleycho** [comment]
LGTM

**Comment by karianna** [comment]
Can't be merged...

**Comment by wesleycho** [comment]
Just needs to be rebased, should be easy.

**Comment by chrisirhc** [comment]
Rebased

**Pull request closed without merging**